src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx

```diff
@@ -715,51 +720,54 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
         poolAsset?.precision ?? 0,
       ),
       // Effectively defined at runtime because of the enabled check below
-      txFeeCryptoBaseUnit: estimatedPoolAssetFeesData?.txFeeCryptoBaseUnit!,
+      txFeeCryptoBaseUnit: poolAssetTxFeeCryptoBaseUnit!,
       // Don't fetch sweep needed if there isn't enough balance for the tx + fees, since adding in a sweep Tx would obviously fail too
       // also, use that as balance checks instead of our current one, at least for the asset (not ROON)
       enabled: Boolean(
-        !!poolAsset?.assetId &&
+        !!poolAsset &&
           bnOrZero(actualAssetDepositAmountCryptoPrecision).gt(0) &&
-          isEstimatedPoolAssetFeesDataSuccess &&
           hasEnoughPoolAssetBalanceForTxPlusFees &&
-          estimatedPoolAssetFeesData?.txFeeCryptoBaseUnit,
+          poolAssetTxFeeCryptoBaseUnit,
       ),
     }),
     [
-      poolAsset?.assetId,
-      poolAsset?.precision,
+      poolAsset,
       poolAssetAccountAddress,
       actualAssetDepositAmountCryptoPrecision,
-      estimatedPoolAssetFeesData?.txFeeCryptoBaseUnit,
-      isEstimatedPoolAssetFeesDataSuccess,
+      poolAssetTxFeeCryptoBaseUnit,
       hasEnoughPoolAssetBalanceForTxPlusFees,
     ],
   )
 
-  const { data: isSweepNeeded, isLoading: isSweepNeededLoading } =
+  const { data: _isSweepNeeded, isLoading: isSweepNeededLoading } =
     useIsSweepNeededQuery(isSweepNeededArgs)
 
+  useEffect(() => {
+    if (_isSweepNeeded !== undefined) setIsSweepNeeded(_isSweepNeeded)
+  }, [_isSweepNeeded])
+
   // Rune balance / gas data and checks
 
   // We reuse lending utils here since all this does is estimating fees for a given deposit amount with a memo
-  // It's not going to be 100% accurate for EVM chains as it doesn't calculate the cost of depositWithExpiry, but rather a simple send,
-  // however that's fine for now until accurate fees estimation is implemented
   const {
     data: estimatedRuneFeesData,
     isLoading: isEstimatedRuneFeesDataLoading,
     isError: isEstimatedRuneFeesDataError,
-    isSuccess: isEstimatedRuneFeesDataSuccess,
   } = useQuoteEstimatedFeesQuery({
     collateralAssetId: thorchainAssetId,
     collateralAccountId: runeAccountId,
     depositAmountCryptoPrecision: actualRuneDepositAmountCryptoPrecision ?? '0',
     confirmedQuote,
   })
 
+  useEffect(() => {
+    if (!estimatedRuneFeesData) return
+    setRuneTxFeeCryptoBaseUnit(estimatedRuneFeesData.txFeeCryptoBaseUnit)
+  }, [estimatedRuneFeesData])
```

Comment:
Why do we need a state field for this since this already exists as `estimatedRuneFeesData.txFeeCryptoBaseUnit`? Or we could always derive it some more with a selector/memo